### PR TITLE
Adding initial OpenShift documentation (v2)

### LINKED ISSUE #1039: Add Openshift/OLM specific installation instructions
**What is the URL of the document?**

**Which section(s) is the issue in?**

I think this needs to be determined. My impulse is to add the instructions as a section under Tutorials like the machine learning example, but I could see it existing under Getting Started I guess, or a new section entirely.

**What needs fixing?**

Adding documentation and determining the appropriate section for the new docs.

**Additional context**

This will be documenting the extra steps needed to set up ACK operators in an Openshift cluster using OLM, i.e. running code-generator to create OLM-specific bundles, installing the operators into Openshift, and setting up single- or multi-tenant configurations. It will also eventually have information on the presence or absence of the AdoptedResource CRD, once we merge some changes to how the AdoptedResource reconciler is handled for OLM installations (see aws-controllers-k8s/runtime#59)

### PR DESCRIPTION
**Pull request by trgeiger**
Version 2 attempt, previous closed PR is #1058

Issue #, if available:
Closes #1039 

Description of changes:
Simplified and aggregated OpenShift-specific stuff into one central document. This seems to flow much better without disrupting any of the existing documentation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/content/docs/user-docs/cross-account-resource-management.md [hunk 1]
```diff
@@ -84,5 +84,4 @@ EOF
 ```
 
 ## Next Steps
-Checkout the [RBAC and IAM permissions overview](../authorization) to understand how ACK manages
-authorization
+Checkout the [RBAC and IAM permissions overview](../authorization) to understand how ACK manages authorization
```
Comment (trgeiger) [inline docs/content/docs/user-docs/cross-account-resource-management.md:87]: Just a dangling newline that I noticed. Can remove this change if anyone prefers typos fixes in separate PRs.

**Comment by trgeiger** [comment]
The new OpenShift page should only have the truly unique stuff to our specific use case for both pre-installation and uninstallation. I left the install instructions in install.md since those seemed to fit nicely and seems like a probably common first-look spot for a user so we can catch them there.

**Review by acornett21 — APPROVED** [review]

File: docs/content/docs/user-docs/install.md [hunk 2]
```diff
@@ -121,6 +121,21 @@ For example, the static manifests needed to install the S3 service controller fo
 [s3-config-dir]: https://github.com/aws-controllers-k8s/s3-controller/tree/main/config
 [s3-repo]: https://github.com/aws-controllers-k8s/s3-controller
 
+
+## Install an ACK service controller with OperatorHub in Red Hat OpenShift
+
+{{% hint type="important" title="Note" %}}
+Please reference the [Openshift](../openshift) page for necessary pre-installataion steps before installing any ACK controllers.
```
Comment (jaypipes) [inline docs/content/docs/user-docs/install.md:128]: s/installataion/installation/ :)

**Review by jaypipes — APPROVED** [review]
Yeah, I like the separate doc for OpenShift prerequisite setup, good call.

File: docs/content/docs/user-docs/install.md (hunk 2 above)
Comment (trgeiger) [reply docs/content/docs/user-docs/install.md]: thanks, fixed :)

**Pull request merged**